pkg/model/iam/iam_builder.go

```diff
@@ -329,7 +334,7 @@ func (b *PolicyBuilder) AddS3Permissions(p *Policy) (*Policy, error) {
 					),
 				})
 			} else {
-				if b.Role == kops.InstanceGroupRoleMaster {
+				if b.Role == kops.InstanceGroupRoleMaster || b.Role == kops.InstanceGroupRoleBastion {
```

Comment:
I would regard S3 access on the bastion as opening up the first tier of the infrastructure to be able to download kubernetes admin certificates. Does not sound like a good idea to me.

Reply:
Did think about this,  master, nodes all get the s3 access role, but bastion does not have the role s3, at the moment need pull the key file from s3, so assign the role to bastion as well.
any better way?

Reply:
What we do in cloudup for aws is upload the ssh key to AWS and use the EC2 service to inject it into the server. 
See more here: upup/pkg/fi/cloudup/awstasks/sshkey.go